### PR TITLE
Authorization bearer

### PR DESCRIPTION
**Pull request by yoshdog**
Thanks to @jacobbednarz for pointing this out.

Reading: 
https://auth0.com/learn/json-web-tokens/

it says:

> Whenever the user wants to access a protected route, it should send the JWT, typically in the Authorization header using the Bearer schema. Therefore the content of the header should look like the following.
>
>Authorization: Bearer <token>

So in this PR we are prefixing `Bearer` in front of the jwt token when setting the Authorization header. We also are supporting requests that have not prefixed a `Bearer` in the Authorization header.

### CODE REVIEW

**Review by Shervanator — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [v2.5.0] - 2019-01-21
+### Changed
+- Ensure we use the JWT Token in the Authorization header using the Bearer schema. We will still support Authoirzation headers without the Bearer schema.
```
Comment (jacobbednarz) [inline CHANGELOG.md]: `s/Authoirzation/Authorization`
```suggestion
- Ensure we use the JWT Token in the Authorization header using the Bearer schema. We will still support Authorization headers without the Bearer schema.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [v2.5.0] - 2019-01-21
```
Comment (jacobbednarz) [inline CHANGELOG.md:7]: To be valid markdown, this should be surrounded by a newline.

```suggestion
## [v2.5.0] - 2019-01-21

```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
...
+### Changed
```
Comment (jacobbednarz) [inline CHANGELOG.md:9]: Same comment as https://github.com/envato/jwt_signed_request/pull/35/files#r249319899

```suggestion
### Changed

```

File: lib/jwt_signed_request/verify.rb [hunk 4]
```diff
@@ -50,8 +50,12 @@ def secret_key
       @secret_key ||= stored_key.fetch(:key) { raise MissingKeyIdError }
     end
 
+    JWT_BEARER_REGEX = /^Bearer ([^*]+)$/
```
Comment (jacobbednarz) [inline lib/jwt_signed_request/verify.rb]: This won't match if `Bearer` is missing. Is that intentional? You might want to swap it to include both.

```rb
\A(Bearer\s)?([^*]+)\z
```

If you don't need to match on the `Bearer` word, I'd suggest swapping `^` for `\A` and `$` for `\z` to perform end of string matches instead of end of line.

Comment (yoshdog) [reply lib/jwt_signed_request/verify.rb]: actually thats much better... thanks

Comment (yoshdog) [reply lib/jwt_signed_request/verify.rb]: updated in: 016c50f
🙇  @jacobbednarz

File: lib/jwt_signed_request/verify.rb [hunk 5]
```diff
@@ -50,8 +50,11 @@ def secret_key
       @secret_key ||= stored_key.fetch(:key) { raise MissingKeyIdError }
     end
 
+    JWT_BEARER_REGEX = /\A(Bearer\s)?([^*]+)\z/
```
Comment (zubin) [inline lib/jwt_signed_request/verify.rb]: Does the trailing whitespace after `Bearer` need to be one character or can it be `\s+`?

File: spec/jwt_signed_request/verify_spec.rb [hunk 6]
```diff
@@ -91,6 +91,16 @@ module JWTSignedRequest
         end
       end
 
+      context 'and was signed with a missing authorization bearer token' do
```
Comment (zubin) [inline spec/jwt_signed_request/verify_spec.rb]: Perhaps the wording could be simpler, eg:

> `when signed without Authorization Bearer syntax`

File: spec/jwt_signed_request/verify_spec.rb [hunk 7]
```diff
@@ -91,6 +91,16 @@ module JWTSignedRequest
...
+        let(:request) do
+          Rack::Request.new(request_env.merge({'HTTP_AUTHORIZATION' => jwt_token}))
+        end
+
+        it 'does not raise any errors' do
```
Comment (zubin) [inline spec/jwt_signed_request/verify_spec.rb:99]: Does this mean it successfully authenticated?

**Review by zubin — APPROVED** [review]
Some nitpicky suggestions but LGTM. 👍

File: lib/jwt_signed_request/verify.rb (hunk 5 above)
Comment (jacobbednarz) [reply lib/jwt_signed_request/verify.rb]: The Bearer authentication scheme is borrowed from the OAuth 2.0 authorization framework ([rfc6750](https://tools.ietf.org/html/rfc6750)) and denotes the format as follows:

```
b64token    = 1*( ALPHA / DIGIT / "-" / "." / "_" / "~" / "+" / "/" ) *"="
credentials = "Bearer" 1*SP b64token
```

In layman terms, the format is "Bearer" + a minimum of 1 space character with an infinite upper bound followed by the base64 encoded value (see [RFC2616, section 2](https://www.w3.org/Protocols/rfc2616/rfc2616-sec2.html) for definitions). While most won't use multiple whitespace conditions, I think @zubin is correct here and we should account for it. Apologies on the oversight.

```suggestion
    JWT_BEARER_REGEX = /\A(Bearer\s+)?([^*]+)\z/
```

File: spec/jwt_signed_request/verify_spec.rb (hunk 7 above)
Comment (yoshdog) [reply spec/jwt_signed_request/verify_spec.rb:99]: yes

**Comment by yoshdog** [comment]
Thanks @zubin @jacobbednarz @Shervanator 

merging it now
